remoteappmanager/db/orm.py

```diff
@@ -129,6 +130,54 @@ def reset(self):
         Base.metadata.create_all(self.engine)
 
 
+class AccountingInterface(ABCAccounting):
+
+    def __init__(self, url, **kwargs):
+        self.db = Database(url, **kwargs)
+
+        # We keep the same session for all transactions
+        # so that the session remains on one thread
+        self.session = self.db.create_session()
+
+    def get_user_by_name(self, user_name):
+        """ Return an orm.User given a user name.  Return None
+        if the user name is not found in the database
+        """
+        with transaction(self.session):
+            return self.session.query(User).filter_by(
+                name=user_name).one_or_none()
+
+    def get_apps_for_user(self, user):
+        """ Return a tuple of tuples, each containing an application
```

Comment:
The results of this method is a state, and should be immutable.  Performance wise, tuples reuse previously used memory block of the same size.